tests/test_admin.py

```diff
@@ -0,0 +1,70 @@
+from pathlib import Path
+
+from django.core.files.base import File
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+import pytest
+from pyquery import PyQuery as pq
+from zds_client.client import ClientError
+
+from zgw_consumers.constants import APITypes
+from zgw_consumers.models import Service
+from zgw_consumers.test import mock_service_oas_get
+
+API_ROOT = "https://example.com/api/v1/"
+OAS_PATH = Path(__file__).parent / "schemas/ztc.yaml"
+TESTS_DIR = Path(__file__).parent
+
+
+@pytest.mark.django_db
+def test_listzaaktypen_mixin(settings, admin_client, requests_mock):
+    """
+    Assert that an error message is displayed in the admin if ClientError is raised
+    """
+    settings.ZGW_CONSUMERS_TEST_SCHEMA_DIRS = [TESTS_DIR / "schemas"]
+    mock_service_oas_get(requests_mock, API_ROOT, "ztc")
+
+    # define errors
+    error_cause = ClientError()
+    error_cause.args = [{"status": 400, "detail": "some other error"}]
+    err = ClientError()
+    err.args = [
+        {
+            "title": "Je hebt geen toestemming om deze actie uit te voeren.",
+            "status": 403,
+            "detail": "Client identifier bestaat niet",
+        }
+    ]
+    err.__cause__ = error_cause
+
+    requests_mock.get(f"{API_ROOT}zaaktypen", exc=err)
+
+    # set up service
+    with open(OAS_PATH, "r") as oas_file:
+        service = Service.objects.create(
+            label="Test",
+            api_type=APITypes.ztc,
+            api_root=API_ROOT,
+            oas_file=File(oas_file, name="schema.yaml"),
```

Comment:
why do you set up the oas_file rather than the oas_url? the mock is being installed that retrieves the OAS, but that codepath should not execute if the OAS file is uploaded - it's either file or URL.